### ISSUE TITLE
Dropout not functioning?

### ISSUE THREAD

**Issue by YilinLiu97** [NifTK/NiftyNet#123]
I may be wrong but I found that the dropout seems not functioning: when dropout was **not used** during training (i.e. keep_prob = 1) but was used solely in the testing, e.g., keep_prob = 0.8, the dice score of the same subject does not change at all compared with the one when dropout was not used in both training and testing, which doesn't seem to make sense (I didn't compare the probabilities, though). Also, dropout in training doesn't seem to help alleviate the overfitting problem (I've tried various values, e.g., keep_prob = 0.5/0.7,etc.).

**Comment by YilinLiu97** [comment]
After doing several experiments, I'm pretty sure that at least the dropout is not functioning at the inference.

**Comment by wyli** [comment]
Hi @YilinLiu97 at the moment dropout is used in [highres3dnet_large](https://github.com/NifTK/NiftyNet/blob/dev/niftynet/network/highres3dnet_large.py#L140) and [dense_vnet](https://github.com/NifTK/NiftyNet/blob/dev/niftynet/network/dense_vnet.py#L245). You can make a 'dropout version' of the other networks by passing `keep_prob` parameter to  [convolution](https://github.com/NifTK/NiftyNet/blob/v0.3.0/niftynet/layer/convolution.py#L161)/[deconvolution layer](https://github.com/NifTK/NiftyNet/blob/v0.3.0/niftynet/layer/deconvolution.py#L199)/[fully_connected](https://github.com/NifTK/NiftyNet/blob/v0.3.0/niftynet/layer/fully_connected.py#L133) layers.

**Comment by somsol** [comment]
Hi @wyli . I've been using a modified version of the dense_vnet (the version before NiftyNer was upgraded to be compatible with tensorflow 1.7), and I've noticed that dropout is applied in the test phase as well (I confirmed this by running the trained network multiple times and getting different outputs each time). Shouldn't dropout be removed during test since we want the results to be consistent?

**Comment by wyli** [comment]
Hi @somsol , new config option `keep_prob` was added to fix the issue (since https://github.com/NifTK/NiftyNet/commit/0e0ba18fbd80309d5713ebda2fc82aebf2f011bb), you'll have to update your network's `layer_op` similarly to https://github.com/NifTK/NiftyNet/commit/0e0ba18fbd80309d5713ebda2fc82aebf2f011bb#diff-7168d2266e42dd1eb1a7a8fa700bf7ffL176
